woocommerce/woocommerce-ios
# Refactor `StoreStatsV4PeriodViewController` with a view model

Part of #5743 


### Description


While working on Home Screen M2 updates with new conversion stats, I started feeling uncomfortable about not having any unit tests on these changes. Store stats are pretty old and we've been adding logic without being able to test them. I decided to take some time to create a view model and move most of the logic that provides view data from the view controller to the view model. My apology for the ~700 diff PR, I tried minimizing the changes but there is a bunch of moved code 😓  The unit tests will be in a separate PR due to the size of this PR. While writing unit tests, I also found some outdated dependency (`currentDate`) and added a new dependency `siteID` to the order and site visit stats predicates.

After this PR, we can also ensure other design updates like for the revenue stats https://github.com/woocommerce/woocommerce-ios/issues/5744 and the time range formatting updates.

Major changes include:

- Created `StoreStatsPeriodViewModel` for `StoreStatsV4PeriodViewController`
- The view model has a publicly settable variable `selectedIntervalIndex` that is set when the user selects a time interval in the chart. Upon changes, all dependent data are recalculated and updated
- Moved all stats text value updates to an observable in the view model: `orderStatsText`, `revenueStatsText`, `visitorStatsText`, `conversionStatsText`
- Other than the stats text values, the following updates are also moved to the view model as an observable:
  - `StatsTimeRangeBarViewModel` which shows the time range text in the header
  - Last updated summary text values
  - Whether the chart should be reloaded (when order stats are updated)
- Removed `currentDate` dependency from `StoreStatsPeriodViewController` and its downstream
- Moved all `ResultsController` storage-level logic from the view controller to the view model, and added `siteID` to the predicates
- Renamed `reloadSiteFields` to `reloadSiteVisitUI`

### Testing instructions


Please make sure the order stats function as before:

- Launch the app
- In the My store tab, tap each time range tab --> the stats values and charts should look as before
- Feel free to try switching to dark/light mode

---
